presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/SqlInvokedFunctionNamespaceManagerConfig.java

```diff
@@ -14,17 +14,39 @@
 package com.facebook.presto.functionNamespace;
 
 import com.facebook.airlift.configuration.Config;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
 import io.airlift.units.Duration;
 import io.airlift.units.MinDuration;
 
+import javax.validation.constraints.NotNull;
+
+import java.util.List;
+
 import static java.util.concurrent.TimeUnit.HOURS;
 import static java.util.concurrent.TimeUnit.MINUTES;
 
 public class SqlInvokedFunctionNamespaceManagerConfig
 {
+    private List<String> servingNamespaces;
     private Duration functionCacheExpiration = new Duration(5, MINUTES);
     private Duration functionInstanceCacheExpiration = new Duration(8, HOURS);
 
+    @NotNull
+    public List<String> getServingNamespaces()
```

Comment:
Instead of duplicating the config in two places, maybe change the `listFunctions` api to `listFunctions(prefix)` and collect that from `FunctionManager` directly.